portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java

```diff
@@ -11,28 +16,71 @@
 @WebServlet("/login")
 public class HomeServlet extends HttpServlet {
   private static final String HOME_PATH = "/";
+  private static final String USER_INFO = "userInfo";
+  private static final String ID = "id";
+  private static final String COMMENT_NICKNAME = "nickname";
+  // Hard-coded email of webpage admin.
+  private static final String ADMIN_EMAIL = "amytn@google.com";
+  private static final String NICKNAME_SERVLET = "/nickname";
+
+  private static String userEmail;
+  private static String nickname;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html");
 
     UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      String currentUserEmail = userService.getCurrentUser().getEmail();
-      String urlToRedirectToAfterUserLogsOut = HOME_PATH;
-      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
 
-      response.getWriter().println("<p>Hello " + currentUserEmail + "!</p>");
-      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+
+    if (!userService.isUserLoggedIn()) {
+      String loginUrl = userService.createLoginURL(HOME_PATH);
```

Comment:
Sorry for not seeing this before but all URL should be Url

Reply:
I think all of mine are Url, do you mean it the other way around?

Reply:
is 'createLoginURL' not your method?

Reply:
createLoginURL isn't my method.